### PR TITLE
Add .NET 9 Target

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
Updates the packages for .NET 9 and includes the following changes:
- Add new .NET 9 for all packages
- Remove deprecated .NET 7 target
- Update package dependencies
- Fix new style, performance, and compilation issues

## Related issues
N/A

## Testing
Existing tests

## [Semver Change](https://github.com/microsoft/healthcare-shared-components/blob/main/docs/Versioning.md)
Breaking

### CODE REVIEW

File: src/Microsoft.Health.Core/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -22,6 +22,6 @@ public static string ComputeHash(this string data)
         EnsureArg.IsNotNull(data, nameof(data));
 
         byte[] hash = SHA256.HashData(Encoding.UTF8.GetBytes(data));
-        return BitConverter.ToString(hash).Replace("-", string.Empty, StringComparison.Ordinal);
```
Comment (wsugarman) [inline src/Microsoft.Health.Core/Extensions/StringExtensions.cs:25]: `BitConverter.ToString(string)` returns values like `AB-CD-EF-01` but `Convert.ToHexString` skips the `-` entirely

File: tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/DelegatingInterfaceImplementationGenerator.cs [hunk 2]
```diff
@@ -136,8 +136,8 @@ private IEnumerable<MemberDeclarationSyntax> GetPropertiesAndMethods()
                             .GetParameters()
                             .Select(p => Parameter(Identifier(p.Name))
                                 .WithType(p.ParameterType.ToTypeSyntax())
-                                .WithModifiers(p.IsDefined(typeof(ParamArrayAttribute), false) ? TokenList(Token(SyntaxKind.ParamsKeyword)) : TokenList())
-                                .WithOptionalAttributeLists(p.CustomAttributes))
+                                .WithModifiers(ParamsTokenList(p))
```
Comment (wsugarman) [inline tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/DelegatingInterfaceImplementationGenerator.cs:139]: These changes are to support signatures like:
```csharp
public void Read(params scoped ReadOnlySpan<string> values)
```

File: test/Microsoft.Health.SqlServer.Tests.E2E/Microsoft.Health.SqlServer.Tests.E2E.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
     <PackageReference Include="Microsoft.IO.RecyclableMemoryStream" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" />
     <PackageReference Include="Newtonsoft.Json" />
+    <PackageReference Include="System.Text.Json" />
```
Comment (arunmk-ms) [inline test/Microsoft.Health.SqlServer.Tests.E2E/Microsoft.Health.SqlServer.Tests.E2E.csproj:18]: Do we need both System.Text.Json and Newtonsoft.Json?

File: .github/workflows/packages-pr.yml [hunk 4]
```diff
@@ -23,7 +23,7 @@ jobs:
         global-json-file: 'global.json'
         dotnet-version: |
           7.x
-          6.x
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply .github/workflows/packages-pr.yml:26]: Yes, this was a mistake!

**Review by arunmk-ms — APPROVED** [review]

**Pull request merged**